templates/lib/yarp/node.rb.erb

```diff
@@ -210,15 +240,19 @@ module YARP
     #
     # def dispatch: (Node) -> void
     def dispatch(root)
-      queue = [root]
+      queue = [[root.enter_event, root]]
 
-      while (node = queue.shift)
-        case node.human
+      # This inline queue avoids having to recursively walk the tree. The queue
+      # contains (event, node) pairs.
+      while (event, node = queue.shift)
+        case event
```

Comment:
We also need to fire a generic `on_node_enter` and `on_node_leave` for all nodes.

Reply:
Let's do that in a follow-up PR, I want to get this merged